packages/multi-entry/src/index.js

```diff
@@ -53,7 +53,7 @@ export default function multiEntry(conf = {}) {
     outputOptions(options) {
       return {
         ...options,
-        entryFileNames: config.entryFileName
+        entryFileNames: (options.preserveModules && config.entryFileName === DEFAULT_OUTPUT) ? options.entryFileName : config.entryFileName
```

Comment:
I dont know if you would prefer this to actually fall back to `options.entryFileNames` or `options.entryFileName`
So raising here to highlight the potential change

```suggestion
        entryFileNames: (options.preserveModules && config.entryFileName === DEFAULT_OUTPUT) ? options.entryFileNames : config.entryFileName
```

Reply:
using entryFileName had a warning so filenames is the go